### PR TITLE
Copy manifest to apps

### PR DESCRIPTION
**Pull request by Hoverbear**
@Xylakant I would appreciate a review on this, as your knowledge exceeds mine on this issue. :)

### CODE REVIEW

File: config/deploy.rb [hunk 1]
```diff
@@ -75,6 +75,19 @@
...
       end
 
+      on roles(:app) do
+        # this needs to be done outside run_locally in order for host to exist
+        remote_dir = "#{host.user}@#{host.hostname}:#{release_path}/public/assets/"
+        execute "mkdir -p #{release_path}/public/assets/"
+        run_locally { execute "rsync -av --delete #{local_dir}/.sprockets-manifest-* #{remote_dir}" }
+
+        # stolen from https://github.com/capistrano/capistrano/blob/master/lib/capistrano/tasks/deploy.rake#L101
+        # we can't directly invoke the task since that runs on deploy roles only
```
Comment (Xylakant) [inline config/deploy.rb]: app is a deploy role, so this code block should not be needed

File: config/deploy.rb [hunk 2]
```diff
@@ -75,6 +75,19 @@
         execute :mv, tmp_current_path, current_path.parent
       end
 
+      on roles(:app) do
+        # this needs to be done outside run_locally in order for host to exist
+        remote_dir = "#{host.user}@#{host.hostname}:#{release_path}/public/assets/"
+        execute "mkdir -p #{release_path}/public/assets/"
+        run_locally { execute "rsync -av --delete #{local_dir}/.sprockets-manifest-* #{remote_dir}" }
```
Comment (Xylakant) [inline config/deploy.rb]: you might also need the manifest.yml

Comment (Hoverbear) [inline config/deploy.rb]: Hm, where would this `manifest.yml` be? 

```
wheelmap@asset:/var/apps/wheelmap/current$ find /var/apps/wheelmap/current/ | grep manifest
/var/apps/wheelmap/current/public/assets/.sprockets-manifest-ca4c4a76cef02a74bbae48b8ba49775d.json
```

Comment (Xylakant) [inline config/deploy.rb]: i was having a  look at the current prod server to compare:

````
felix@spiderman:/var/apps/wheelmap/production/current/public/assets$ ls -la manifest.yml
-rw-rw-r-- 1 rails rails 80708 Jun  7 17:27 manifest.yml
````

File: config/deploy.rb (hunk 1 above)
Comment (Hoverbear) [inline config/deploy.rb]: Removed in ac13761

File: config/deploy.rb (hunk 2 above)
Comment (Xylakant) [inline config/deploy.rb]: yes, looks like the manifest.yml is a rails3 artefact.

Comment (Hoverbear) [inline config/deploy.rb]: I manually rsynced a copy over and it seems to be working partially. 

screen shot 2016-09-06 at 14 26 09

**Comment by Xylakant** [comment]
cool, can you do a test deployment?

**Comment by Xylakant** [comment]
did you change your /etc/hosts to point to wheelmap to the staging lb?

**Comment by Hoverbear** [comment]
Yes I changed my hostfile.

**Comment by Xylakant** [comment]
the app on staging expects the hostname to be staging.wheelmap.org, not wheelmap.org

**Comment by Hoverbear** [comment]
Ah that might be the reason.

**Comment by Hoverbear** [comment]
I'm rebuilding the cluster now. Should I point the capistrano files at this branch or wait for merge before deploying the apps?

**Comment by Xylakant** [comment]
why rebuild the cluster?

**Comment by Hoverbear** [comment]
> cool, can you do a test deployment?

I was under the impression this was a full deployment end-to-end since there were changes in regards to consul as well as this app.

**Comment by Xylakant** [comment]
we need to be able to roll out partial changes without rebuilding the whole infrastructure. We can't take down prod just because we changed a host name

**Comment by Hoverbear** [comment]
https://github.com/sozialhelden/chef-repository/issues/121 will accomplish this.

**Comment by Xylakant** [comment]
so currently all you need to do is run knife solo manually for all machines, but rebuilding shouldn't be needed.

**Comment by Hoverbear** [comment]
**Please do not merge this until I remove the next commit, it is for testing deploy only**

**Comment by Hoverbear** [comment]
This fixes all but some scss imports. I'll be making a seperate PR for this.

screen shot 2016-09-07 at 10 43 26

**Pull request merged**